metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_config.py

```diff
@@ -0,0 +1,197 @@
+
+    @validator("authentication_method")
+    def validate_auth_method(cls, value):
+        allowed_methods = ["password", "PAT"]
+        if value not in allowed_methods:
+            raise ValueError(
+                f"authentication_method must be one of {allowed_methods}",
+            )
+        return value
+
+    @validator("password")
+    def validate_password(cls, value, values):
+        if values.get("authentication_method") == "PAT" and not value:
+            raise ValueError(
+                "Password (Personal Access Token) is required when using PAT authentication",
+            )
+        return value
+
+
+class ProfileConfig(GEProfilingConfig):
+    partition_profiling_enabled: bool = Field(
+        default=False,
+        description="Partition profiling disabled for Dremio.",
+    )
+    include_field_median_value: bool = Field(
+        default=False,
+        description="Median causes a number of issues in Dremio.",
+    )
+    query_timeout: int = Field(
+        default=300, description="Time before cancelling Dremio profiling query"
+    )
+
+    row_count: bool = True
+    column_count: bool = True
+    sample_values: bool = True
+
+
+class DremioSourceMapping(EnvConfigMixin, ConfigModel):
+    platform: Optional[str] = Field(
+        default=None,
+        description="Source connection made by Dremio (e.g. S3, Snowflake)",
+    )
+    platform_name: Optional[str] = Field(
+        default=None,
+        description="Alias of platform in Dremio connection",
+    )
+    platform_instance: Optional[str] = Field(
+        default=None,
+        description="Platform instance of source connection in Datahub",
+    )
+    dremio_source_type: Optional[str] = Field(
+        default=None,
+        description="Source connection made by Dremio (e.g. S3, Snowflake)",
+    )
+
+    root_path: Optional[str] = Field(
+        default=None,
+        description="Root path of source - Extracted from Dremio API",
+        hidden_from_docs=True,
+    )
+    database_name: Optional[str] = Field(
+        default=None,
+        description="Database of source - Extracted from Dremio API",
+        hidden_from_docs=True,
+    )
+
+
+class DremioSourceConfig(
+    DremioConnectionConfig,
+    StatefulIngestionConfigBase,
+    EnvConfigMixin,
+    PlatformInstanceConfigMixin,
+):
+
+    domain: Optional[str] = Field(
+        default=None,
+        description="Domain for all source objects.",
+    )
+
+    source_mappings: Optional[List[DremioSourceMapping]] = Field(
+        default=None,
+        description="Mappings from Dremio sources to DataHub platforms and datasets.",
+    )
+
+    # Entity Filters
+    schema_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for schemas to filter",
+    )
+    dataset_pattern: AllowDenyPattern = Field(
+        default=AllowDenyPattern.allow_all(),
+        description="Regex patterns for schemas to filter",
+    )
```

Comment:
Can you elaborate in description what exactly does `schema_pattern` and `dataset_pattern` filter ? Spaces, Sources, Folders, Tables, Views ? What format does this allowdenypattern string expect ? AFAIK, schema is not a native term in dremio.

Reply:
`dataset_pattern` is for the tables and views (PDS/VDS)
`schema_pattern` is for the sources, spaces and folders. Although it is not a native term within Dremio it will be familiar to those consuming downstream as the most common ways are through ODBC and JDBC. Any tools that connect to Dremio see it as a DB, so I think the DB terminology is fine.

Reply:
Is `dataset_pattern` expected to be fully qualified (`<source>.<folder>.<table name>`) or is it matched against just table name (`<table name>`). It would be nice to document this. 

Would be nice to document how it works for `schema_pattern` as well. Refer description of table_pattern [here](https://datahubproject.io/docs/generated/ingestion/sources/snowflake/#config-details:~:text=share%20is%20created.-,table_pattern,-AllowDenyPattern) for example.

If its not already using qualified name, I would recommend matching it against qualified name as that's what we do for rest of sources and it allows full flexibility for user in terms of specifying patterns.


Based on example from dremio [doc](https://docs.dremio.com/24.3.x/reference/api/catalog/table)  -> the container of table or view could be multi-level path and not some fixed number of levels. Curious how schema looks like for such cases.

Reply:
Looks like schema_pattern uses `TABLE_SCHEMA` column which is fully qualified so this is fine.